addons/web/static/src/views/list/list_controller.js

```diff
@@ -237,12 +239,7 @@ export class ListController extends Component {
     }
 
     onClickDiscard() {
-        const editedRecord = this.model.root.editedRecord;
-        if (editedRecord.isNew) {
-            this.model.root.removeRecord(editedRecord);
-        } else {
-            editedRecord.discard();
-        }
+        this.model.root.leaveEditMode({ discard: true });
```

Comment:
any reason why `leaveEditMode` is not on model itself? Also, i am not sure I like the method name. Shouldn't it be `setReadonly` or something that explains the goal of the method?